### PR TITLE
tests: e2e test for HPA component (PROJQUAY-3408)

### PR DESCRIPTION
**Pull request by nichitagutu**
added e2e test for HPA component

### CODE REVIEW

File: e2e/hpa/01-unmanage-hpa.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  - kind: route
+    managed: true
+  - kind: mirror
+    managed: true
+  - kind: monitoring
+    managed: true
+  - kind: tls
+    managed: true
+---
+apiVersion: autoscaling/v1
```
Comment (ricardomaraschini) [inline e2e/hpa/01-unmanage-hpa.yaml:31]: Why do we create this HPA here ?

File: e2e/hpa/00-assert.yaml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    status: "True"
+    type: Available
+  - message: All registry components created
+    reason: ComponentsCreationSuccess
+    status: "True"
+    type: ComponentsCreated
+  - message: All objects created/updated successfully
+    reason: ComponentsCreationSuccess
+    status: "False"
+    type: RolloutBlocked
```
Comment (ricardomaraschini) [inline e2e/hpa/00-assert.yaml:88]: I think that this assert should contain the HPA as well. The HPA creation is expected, no?

File: e2e/hpa/01-assert.yaml [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+    type: ComponentsCreated
+  - message: All objects created/updated successfully
+    reason: ComponentsCreationSuccess
+    status: "False"
+    type: RolloutBlocked
+---
+apiVersion: autoscaling/v1
+kind: HorizontalPodAutoscaler
+metadata:
+  name: hpa-quay-mirror
```
Comment (ricardomaraschini) [inline e2e/hpa/01-assert.yaml:110]: This does not seem right. When the QuayRegistry is created we expect to see three HPAs and their names are:

```
X-clair-app
X-quay-app
X-quay-mirror
```
Where X is equal to QuayRegistry.Spec.Name.

File: e2e/hpa/02-assert.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+kind: HorizontalPodAutoscaler
+metadata:
+  annotations:
+    quay-component: mirror
+  labels:
+    quay-operator/quayregistry: hpa
+status:
+  currentReplicas: 2
+  desiredReplicas: 2
+---
```
Comment (ricardomaraschini) [inline e2e/hpa/02-assert.yaml:114]: This one seems to be wrong as well. We do not specify any name for the HPA so we don't know what HPA is used to assert.

**Review by ricardomaraschini — CHANGES_REQUESTED** [review]

File: e2e/hpa/00-assert.yaml (hunk 2 above)
Comment (nichitagutu) [reply e2e/hpa/00-assert.yaml:88]: Doesn't the first message assume that the HPA is created?

`message: Horizontal pod autoscaler found`

> I think that this assert should contain the HPA as well. The HPA creation is expected, no?

Comment (ricardomaraschini) [reply e2e/hpa/00-assert.yaml:88]: Yep, that makes sense.

File: e2e/hpa/01-assert.yaml (hunk 3 above)
Comment (nichitagutu) [reply e2e/hpa/01-assert.yaml:110]: I've thought that testing against 1 hpa would be enough, would write all 3 of them then.

> This does not seem right. When the QuayRegistry is created we expect to see three HPAs and their names are:
> 
> ```
> X-clair-app
> X-quay-app
> X-quay-mirror
> ```
> 
> Where X is equal to QuayRegistry.Spec.Name.

File: e2e/hpa/02-assert.yaml (hunk 4 above)
Comment (nichitagutu) [reply e2e/hpa/02-assert.yaml:114]: True, name is not specified, but doesn't annotations and labels specify which hpa the test asserts (hpa-quay-mirror in this case) or there could be more than one hpa per component?
```
  annotations:
    quay-component: mirror
  labels:
    quay-operator/quayregistry: hpa

File: e2e/hpa/01-unmanage-hpa.yaml (hunk 1 above)
Comment (nichitagutu) [reply e2e/hpa/01-unmanage-hpa.yaml:31]: > Why do we create this HPA here ?

I'm not sure what creation you're mentioning.

```
apiVersion: autoscaling/v1
kind: HorizontalPodAutoscaler
metadata:
  name: hpa-quay-mirror
spec:
  minReplicas: 1
  ```
  If you mean this, these lines update the existing hpa, existence of which is asserted in the first step.

**Comment by ricardomaraschini** [comment]
@Nikita757 https://github.com/quay/quay-operator/pull/671 was merged so you gonna need to update references to "base" component to "quay" for this one to get in. Let me know when I can give this one a final review.

**Comment by nichitagutu** [comment]
@ricardomaraschini Updated the references

File: e2e/hpa/00-assert.yaml [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+---
+apiVersion: kuttl.dev/v1beta1
+kind: TestAssert
```
Comment (ricardomaraschini) [inline e2e/hpa/00-assert.yaml:91]: Why are we adding this TestAssert here?

**Review by ricardomaraschini — COMMENTED** [review]
This looks awesome! There is one final question then we can merge this. See my comment.

**Comment by nichitagutu** [comment]
> This looks awesome! There is one final question then we can merge this. See my comment.
> Why are we adding this TestAssert here?

I've added it to increase timeout for this particular test, because otherwise 500 seconds may not be enough for everything to get up and running.

**Review by ricardomaraschini — APPROVED** [review]

**Pull request merged**